src/System.Private.CoreLib/shared/System/CustomAttribute.cs

```diff
@@ -0,0 +1,472 @@
+                    return typeof(byte);
+
+                case (CustomAttributeEncoding.Int16):
+                    return typeof(short);
+
+                case (CustomAttributeEncoding.UInt16):
+                    return typeof(ushort);
+
+                case (CustomAttributeEncoding.UInt32):
+                    return typeof(uint);
+
+                case (CustomAttributeEncoding.Int64):
+                    return typeof(long);
+
+                case (CustomAttributeEncoding.UInt64):
+                    return typeof(ulong);
+
+                case (CustomAttributeEncoding.Float):
+                    return typeof(float);
+
+                case (CustomAttributeEncoding.Double):
+                    return typeof(double);
+
+                case (CustomAttributeEncoding.Object):
+                    return typeof(object);
+
+                default:
+                    throw new ArgumentException(SR.Format(SR.Arg_EnumIllegalVal, (int)encodedType), nameof(encodedType));
+            }
+        }
+
+        private static object EncodedValueToRawValue(long val, CustomAttributeEncoding encodedType)
+        {
+            switch (encodedType)
+            {
+                case CustomAttributeEncoding.Boolean:
+                    return (byte)val != 0;
+
+                case CustomAttributeEncoding.Char:
+                    return (char)val;
+
+                case CustomAttributeEncoding.Byte:
+                    return (byte)val;
+
+                case CustomAttributeEncoding.SByte:
+                    return (sbyte)val;
+
+                case CustomAttributeEncoding.Int16:
+                    return (short)val;
+
+                case CustomAttributeEncoding.UInt16:
+                    return (ushort)val;
+
+                case CustomAttributeEncoding.Int32:
+                    return (int)val;
+
+                case CustomAttributeEncoding.UInt32:
+                    return (uint)val;
+
+                case CustomAttributeEncoding.Int64:
+                    return (long)val;
+
+                case CustomAttributeEncoding.UInt64:
+                    return (ulong)val;
+
+                case CustomAttributeEncoding.Float:
+                    unsafe { return *(float*)&val; }
+
+                case CustomAttributeEncoding.Double:
+                    unsafe { return *(double*)&val; }
+
+                default:
+                    throw new ArgumentException(SR.Format(SR.Arg_EnumIllegalVal, (int)val), nameof(val));
+            }
+        }
+        #endregion
+
+        private readonly object m_value;
+        private readonly Type m_argumentType;
+
+        #region Constructor
+        public CustomAttributeTypedArgument(Type argumentType, object value)
+        {
+            // value can be null.
+            if (argumentType == null)
+                throw new ArgumentNullException(nameof(argumentType));
+
+            m_value = (value == null) ? null : CanonicalizeValue(value);
```

Comment:
use `if (value != null) m_value = CanonicalizeValue(value);`

Reply:
it's a struct, so all fields must be assigned, your snippet won't compile.